platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/sources/LegacyFileConfigSource.java

```diff
@@ -47,16 +47,14 @@ public class LegacyFileConfigSource extends AbstractConfigSource {
 
     /**
      * Creates an instance that provides the config properties from the legacy settings.txt file
-     *
+     * @param settingsPath the path to the settings.txt file
      * @return config source for the settings.txt file
      * @throws IOException if settings.txt can not be loaded
      */
-    public static LegacyFileConfigSource ofSettingsFile() throws IOException {
-        final Path path = FileSystems.getDefault()
-                .getPath("settings.txt")
-                .toAbsolutePath()
-                .normalize();
-        return new LegacyFileConfigSource(path, ConfigSourceOrdinalConstants.LEGACY_PROPERTY_FILE_ORDINAL_FOR_SETTINGS);
+    @NonNull
+    public static LegacyFileConfigSource ofSettingsFile(@NonNull final Path settingsPath) throws IOException {
```

Comment:
why not have both methods?

Reply:
Mostly just because we don't have any current use cases for the zero arg constructor, and since this is a legacy type I thought we'd try to avoid new use cases in the future. I can add it back in if you feel strongly about it.